.github/workflows/main.yml

```diff
@@ -0,0 +1,253 @@
+        if: ${{ github.event_name != 'release' }}
+        run: |
+          time=$(date +%s)
+          ref="${{ github.head_ref || github.ref_name }}"
+          ref_clean="${ref/\//-}"
+          suffix="ci-$time-$ref_clean-${{ github.run_id }}"
+          echo "version=0.0.0-$suffix" >> $GITHUB_OUTPUT
+
+      - name: Publish API
+        run: >
+          dotnet publish src/Api/
+          --no-restore
+          --no-build
+          --configuration Release
+          --runtime win-x86
+          --no-self-contained
+          --output publish/Api/
+          -p:ContinuousIntegrationBuild=true
+          ${{ steps.prerelease-version.outputs.version && format('-p:Version={0}', steps.prerelease-version.outputs.version) || '' }}
+
+      - name: Publish AdminConsole
+        run: >
+          dotnet publish src/AdminConsole/
+          --no-restore
+          --no-build
+          --configuration Release
+          --runtime win-x86
+          --no-self-contained
+          --output publish/AdminConsole/
+          -p:ContinuousIntegrationBuild=true
+          ${{ steps.prerelease-version.outputs.version && format('-p:Version={0}', steps.prerelease-version.outputs.version) || '' }}
+
+      - name: Upload API artifacts
+        uses: actions/upload-artifact@a8a3f3ad30e3422c9c7b888a15615d19a852ae32 # v3.1.3
+        with:
+          name: Api
+          path: publish/Api/
+
+      - name: Upload AdminConsole artifacts
+        uses: actions/upload-artifact@a8a3f3ad30e3422c9c7b888a15615d19a852ae32 # v3.1.3
+        with:
+          name: AdminConsole
+          path: publish/AdminConsole/
+
+  # Deploy the apps
+  deploy:
+    needs:
+      # Technically, it's not required for the format job to succeed for us to push the package,
+      # so we may consider removing it as a prerequisite here.
+      - format
+      - test
+      - pack
+    runs-on: ubuntu-latest
+    environment: devtest
+    permissions:
+      actions: read
+      packages: write
+
+    steps:
+      - name: Download API artifacts
+        uses: actions/download-artifact@9bc31d5ccc31df68ecc42ccf4149144866c47d8a # v3.0.2
+        with:
+          name: Api
+
+      - name: Download AdminConsole artifacts
+        uses: actions/download-artifact@9bc31d5ccc31df68ecc42ccf4149144866c47d8a # v3.0.2
+        with:
+          name: AdminConsole
+
+      - name: Archive API
+        run: zip -r Api.zip publish/Api/
+
+      - name: Archive AdminConsole
+        run: zip -r AdminConsole.zip publish/AdminConsole/
+
+      - name: Login to Azure
+        uses: Azure/login@1f63701bf3e6892515f1b7ce2d2bf1708b46beaf  # v1.4.3
+        with:
+          creds: ${{ secrets.APP_SERVICE_SP }}
+
+      - name: Configure app settings
+        # Temporary
+        if: ${{ false }}
+        run: >
+          az webapp config appsettings set
+          --resource-group ${{ secrets.AZURE_RESOURCE_GROUP_NAME }}
+          --name ${{ secrets.AZURE_APP_NAME }}
+          --slot staging
```

Comment:
`DD_VERSION` isn't being set here (it's a bit more effort to retrieve the actual version in this setup). Do we want to continue setting it like this or find a better way (i.e. resolve in code instead)?

Reply:
I'd love to find a better way - I have a ticket open with DataDog to figure it out, but not sure if possible

Reply:
We would probably need to still set it here, will update if I hear another solution

Reply:
Worst case, I'd set the variable on program startup since it's easier to retrieve version there.

Reply:
Looks like you're already setting the version in code here:

<img width="640" alt="image" src="https://github.com/passwordless/passwordless-server/assets/1935960/548cb556-131e-4242-b108-19fc3528bedf">